diff --git a/includes/discovery/sensors/pre-cache/nokia-isam.inc.php b/includes/discovery/sensors/pre-cache/nokia-isam.inc.php
index 1cbd26099eb4c900e1ba791fa64d2dc19cca5b97..f5426942eef3b0314b1972f205eff5f2d15124e8 100644
--- a/includes/discovery/sensors/pre-cache/nokia-isam.inc.php
+++ b/includes/discovery/sensors/pre-cache/nokia-isam.inc.php
@@ -3,7 +3,7 @@
 // Lookup the Isam Port Numbering Method
 $asamPortNumberingMethod = snmp_getnext($device, 'asamPortNumberingMethodSystem', '-Ovq', 'ASAM-SYSTEM-MIB', 'nokia');
 $eqptHolderActualType = snmp_get($device, 'eqptHolderActualType.17', '-Ovq', 'ASAM-EQUIP-MIB', 'nokia');
-echo 'ISAM Slot/Port Mapping for '.$eqptHolderActualType.'';
+echo 'ISAM Slot/Port Mapping for ' . $eqptHolderActualType . '';
 
 //Typ to prodcut name
 $eqptHolderTable = [
@@ -17,10 +17,10 @@ $eqptHolderTable = [
 ];
 
 if ($asamPortNumberingMethod == 'positionBasedSlotId') {
-// Slot mapping for "positionBasedSlotId" (see ASAM-SYSTEM-MIB)
-// also hardcoded because of no reference found in MIB but matched on DSLAM typ (eqptHolderActualTyp)
-// acutal tested on ISAM 7330/56/60/62
-// Nokia IDs for ISAM 7330
+    // Slot mapping for "positionBasedSlotId" (see ASAM-SYSTEM-MIB)
+    // also hardcoded because of no reference found in MIB but matched on DSLAM typ (eqptHolderActualTyp)
+    // acutal tested on ISAM 7330/56/60/62
+    // Nokia IDs for ISAM 7330
     if ($eqptHolderActualType == 'NFXS-B') {
         $slotTable = [
             '4352' => 'ntio-1:',
@@ -51,8 +51,8 @@ if ($asamPortNumberingMethod == 'positionBasedSlotId') {
             '28929' => 'ctrl:7/1:',
             '28931' => 'lt:7/1/1:',
             '28932' => 'lt:7/1/2:',
-	];
-// Slot IDs for Nokia ISAM 7356
+        ];
+        // Slot IDs for Nokia ISAM 7356
     } elseif ($eqptHolderActualType == 'NFXR-A') {
         $slotTable = [
             '4353' => 'nt:',
@@ -73,8 +73,8 @@ if ($asamPortNumberingMethod == 'positionBasedSlotId') {
             '28929' => 'ctrl:7/1:',
             '28931' => 'lt:7/1/1:',
             '28932' => 'lt:7/1/2:',
-	];
-// Slot IDs for Nokia ISAM 7360
+        ];
+        // Slot IDs for Nokia ISAM 7360
     } elseif ($eqptHolderActualType == 'NFXS-F') {
         $slotTable = [
             '4352' => 'acu:1/1:',
@@ -106,30 +106,31 @@ if ($asamPortNumberingMethod == 'positionBasedSlotId') {
             '28929' => 'ctrl:7/1:',
             '28931' => 'lt:7/1/1:',
             '28932' => 'lt:7/1/2:',
-	];
-// Slot IDs for Nokia ISAM 7362
+        ];
+        // Slot IDs for Nokia ISAM 7362
     } elseif ($eqptHolderActualType == 'CFXR-A') {
         $slotTable = [
             '4353' => 'nt:',
             '4355' => 'lt:1/1/1:',
-        ]; }
+        ];
+    }
 } else {
-// Slot mapping
-// FIXME Hardcoded to Num Based Slot Numbering. This should be pulled from SNMP
-$slotTable = [
-    '4352' => 'acu:1/1/',
-    '4353' => 'nt-a:',
-    '4354' => 'nt-b:',
-    '4355' => 'lt:1/1/1/',
-    '4356' => 'lt:1/1/2/',
-    '4357' => 'lt:1/1/3/',
-    '4358' => 'lt:1/1/4/',
-    '4359' => 'lt:1/1/5/',
-    '4360' => 'lt:1/1/6/',
-    '4361' => 'lt:1/1/7/',
-    '4362' => 'lt:1/1/8/',
-    '4481' => '4481', // FIXME define this
-];
+    // Slot mapping
+    // FIXME Hardcoded to Num Based Slot Numbering. This should be pulled from SNMP
+    $slotTable = [
+        '4352' => 'acu:1/1/',
+        '4353' => 'nt-a:',
+        '4354' => 'nt-b:',
+        '4355' => 'lt:1/1/1/',
+        '4356' => 'lt:1/1/2/',
+        '4357' => 'lt:1/1/3/',
+        '4358' => 'lt:1/1/4/',
+        '4359' => 'lt:1/1/5/',
+        '4360' => 'lt:1/1/6/',
+        '4361' => 'lt:1/1/7/',
+        '4362' => 'lt:1/1/8/',
+        '4481' => '4481', // FIXME define this
+    ];
 }
 $pre_cache['nokiaIsamSlot'] = snmpwalk_cache_multi_oid($device, 'eqptBoardContainerOffset', [], 'ASAM-EQUIP-MIB', 'nokia');
 $pre_cache['nokiaIsamSlot'] = snmpwalk_cache_multi_oid($device, 'eqptBoardIfSlotId', $pre_cache['nokiaIsamSlot'], 'ASAM-EQUIP-MIB', 'nokia');
diff --git a/includes/discovery/sensors/state/nokia-isam.inc.php b/includes/discovery/sensors/state/nokia-isam.inc.php
index dbc8b5669f1e58493f89d878b211e94c2037e29e..58bd97fcd31a2adcd70eeadf646389fefcb8282c 100644
--- a/includes/discovery/sensors/state/nokia-isam.inc.php
+++ b/includes/discovery/sensors/state/nokia-isam.inc.php
@@ -6,12 +6,12 @@ foreach ($snmp_data['nokiaIsamEqpBoardTable'] as $index => $data) {
     if (is_array($data)) {
         $oid = '.1.3.6.1.4.1.637.61.1.23.3.1.7.' . $index;
         $state_name = 'eqptBoardOperError';
-	$current = $data['eqptBoardOperError'];
-	// Skip empty and not planned boards / false table entrys
-        if ($current == 0 OR $data['eqptSlotActualType'] == 'EMPTY' AND $data['eqptSlotPlannedType'] == 'NOT_PLANNED') {
-		continue;
-	}
-	$descr = $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] . ' ' . $data['eqptSlotActualType'] . ' (' . $data['eqptSlotPlannedType'] . ')';
+        $current = $data['eqptBoardOperError'];
+        // Skip empty and not planned boards / false table entrys
+        if ($current == 0 or $data['eqptSlotActualType'] == 'EMPTY' and $data['eqptSlotPlannedType'] == 'NOT_PLANNED') {
+            continue;
+        }
+        $descr = $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] . ' ' . $data['eqptSlotActualType'] . ' (' . $data['eqptSlotPlannedType'] . ')';
         $states = [
             ['value' => 1, 'generic' => 0, 'graph' => 0, 'descr' => 'no-error'],
             ['value' => 2, 'generic' => 1, 'graph' => 0, 'descr' => 'type-mismatch'],
@@ -36,19 +36,19 @@ foreach ($snmp_data['nokiaIsamEqpBoardTable'] as $index => $data) {
             ['value' => 255, 'generic' => 2, 'graph' => 0, 'descr' => 'unknown-error'],
         ];
         create_state_index($state_name, $states);
-	if ($pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'ctrl:2/1:' OR $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:2/1/1:' OR $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:2/1/2:') {
+        if ($pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'ctrl:2/1:' or $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:2/1/1:' or $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:2/1/2:') {
             $group = 'Remote Shelf: 2';
-	} elseif ($pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'ctrl:3/1:' OR $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:3/1/1:' OR $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:3/1/2:') {
+        } elseif ($pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'ctrl:3/1:' or $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:3/1/1:' or $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:3/1/2:') {
             $group = 'Remote Shelf: 3';
-	} elseif ($pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'ctrl:4/1:' OR $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:4/1/1:' OR $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:4/1/2:') {
+        } elseif ($pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'ctrl:4/1:' or $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:4/1/1:' or $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:4/1/2:') {
             $group = 'Remote Shelf: 4';
-	} elseif ($pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'ctrl:5/1:' OR $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:5/1/1:' OR $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:5/1/2:') {
+        } elseif ($pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'ctrl:5/1:' or $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:5/1/1:' or $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:5/1/2:') {
             $group = 'Remote Shelf: 5';
-	} elseif ($pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'ctrl:6/1:' OR $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:6/1/1:' OR $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:6/1/2:') {
+        } elseif ($pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'ctrl:6/1:' or $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:6/1/1:' or $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:6/1/2:') {
             $group = 'Remote Shelf: 6';
-	} elseif ($pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'ctrl:7/1:' OR $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:7/1/1:' OR $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:7/1/2:') {
+        } elseif ($pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'ctrl:7/1:' or $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:7/1/1:' or $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:7/1/2:') {
             $group = 'Remote Shelf: 7';
-	} elseif ($pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'ctrl:8/1:' OR $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:8/1/1:' OR $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:8/1/2:') {
+        } elseif ($pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'ctrl:8/1:' or $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:8/1/1:' or $pre_cache['nokiaIsamSlot'][$index]['numBasedSlot'] == 'lt:8/1/2:') {
             $group = 'Remote Shelf: 8';
         } else {
             $group = $pre_cache['nokiaProductName'];
